ee/src/main/java/org/jboss/as/ee/component/deployers/ModuleJndiBindingProcessor.java

```diff
@@ -167,6 +172,10 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                 if(config.isInvalid()) {
                     continue;
                 }
+                ClassInfo classInfo = index.getClassByName(DotName.createSimple(config.getClassName()));
```

Comment:
To ease future maintenance this needs a nice comment before it explaining why this will work; i.e. an Index contains info on all classes in a jar, and the CompositeIndex in the 'index' var aggregates all the Index objects for the jars visible to the deployment unit. So you can scan it for relationships between classes and not miss out on any.